src/script/util/util.js

```diff
@@ -302,15 +303,25 @@ markdownit.renderer.rules.link_open = (tokens, idx, options, env, self) => {
   const link = tokens[idx];
   const href = link.attrGet('href');
   const isEmail = href.startsWith('mailto:');
+  const nextToken = tokens[idx + 1];
+  const text = nextToken && nextToken.type === 'text' ? nextToken.content : '';
+  if (!href || !text.trim()) {
+    nextToken.content = '';
+    const closeToken = tokens.slice(idx).find(token => token.type === 'link_close');
+    closeToken.type = 'text';
+    closeToken.content = '';
+    return `[${text}](${href})`;
+  }
   if (isEmail) {
-    const email = href.replace(/^mailto:/, '');
+    const email = SanitizationUtil.escapeString(href.replace(/^mailto:/, ''));
     link.attrPush(['onclick', `z.util.SanitizationUtil.safeMailtoOpen(event, '${email}')`]);
   } else {
     link.attrPush(['target', '_blank']);
     link.attrPush(['rel', 'nofollow noopener noreferrer']);
   }
   if (link.markup !== 'linkify') {
-    link.attrPush(['class', 'dangerLink']);
+    link.attrPush(['class', 'markdown-link']);
```

Comment:
Let's try to use dataset (if allowed by markdownit). This way we know it not for styling and we won't be tempted to remove it in the future.